lib/srv/app/azure/handler.go

```diff
@@ -285,7 +290,16 @@ func (s *handler) getToken(ctx context.Context, managedIdentity string, scope st
 	timeoutCtx, cancel := context.WithTimeout(ctx, getTokenTimeout)
 	defer cancel()
 
-	return utils.FnCacheGet(timeoutCtx, s.tokenCache, key, func(ctx context.Context) (*azcore.AccessToken, error) {
+	token, err := utils.FnCacheGet(timeoutCtx, s.tokenCache, key, func(ctx context.Context) (*azcore.AccessToken, error) {
 		return s.getAccessToken(ctx, managedIdentity, scope)
 	})
+
+	if err != nil {
+		if timeoutCtx.Err() == err {
```

Comment:
This condition looks wield at first glance. 

Can we just check err.Is(err, context.DeadlineExceeded) ?

Reply:
The difference is that we check that _our_ timeout has expired, whereas `err.Is(err, context.DeadlineExceeded)` checks that _some_ timeout has expired.

Reply:
After refactoring to use `clock.After` the offending piece of code is gone.